docs/minikube-tutorial.md

```diff
@@ -0,0 +1,196 @@
+
+## Table of Contents
+
+- [Start a Kubernetes cluster with minikube](#start-a-kubernetes-cluster-with-minikube)
+  * [What is minikube?](#what-is-minikube)
+  * [Install minikube and start a Kubernetes cluster](#install-minikube-and-start-a-kubernetes-cluster)
+  * [Install kubectl to access the cluster](#install-kubectl-to-access-the-cluster)
+- [Install TiDB operator and run a TiDB cluster with it](#install-tidb-operator-and-run-a-tidb-cluster-with-it)
+  * [Install helm](#install-helm)
+  * [Install TiDB operator in the Kubernetes cluster](#install-tidb-operator-in-the-kubernetes-cluster)
+  * [Launch a TiDB cluster](#launch-a-tidb-cluster)
+  * [Test TiDB cluster](#test-tidb-cluster)
+  * [Delete TiDB cluster](#delete-tidb-cluster)
+- [FAQs](#faqs)
+  * [TiDB cluster in minikube is not responding or responds slow](#tidb-cluster-in-minikube-is-not-responding-or-responds-slow)
+
+## Start a Kubernetes cluster with minikube
+
+### What is minikube?
+
+[Minikube](https://kubernetes.io/docs/setup/minikube/) can start a local
+Kubernetes cluster inside a VM on your laptop. It works on macOS, Linux, and
+Windows.
+
+### Install minikube and start a Kubernetes cluster
+
+See [Installing Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) to install
+minikube on your machine.
+
+After you installed minikube, you can run the following command to start a
+Kubernetes cluster.
+
+```
+minikube start
+```
+
+For Chinese mainland users, you may use local gcr.io mirrors such as
+`registry.cn-hangzhou.aliyuncs.com/google_containers`.
+
+```
+minikube start --image-repository registry.cn-hangzhou.aliyuncs.com/google_containers
+```
+
+or configure HTTP/HTTPS proxy environments in your docker, e.g.
+
+```
+# change 127.0.0.1:1086 to your http/https proxy server IP:PORT
+minikube start --docker-env https_proxy=http://127.0.0.1:1086 \
+  --docker-env http_proxy=http://127.0.0.1:1086
+```
+
+See [minikube setup](https://kubernetes.io/docs/setup/minikube/) for more options to
+configure your virtual machine and Kubernetes cluster.
+
+### Install kubectl to access the cluster
+
+The Kubernetes command-line tool,
+[kubectl](https://kubernetes.io/docs/user-guide/kubectl/), allows you to run
+commands against Kubernetes clusters.
+
+Install kubectl according to the instructions in [Install and Set Up kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+
+After kubectl is installed, test your minikube Kubernetes cluster:
+
+```
+kubectl cluster-info
+```
+
+## Install TiDB operator and run a TiDB cluster with it
+
+### Install helm
+
+Helm is the package manager for Kubernetes and is what allows us to install all of the distributed components of TiDB in a single step. Helm requires both a server-side and a client-side component to be installed.
+
+```
+curl https://raw.githubusercontent.com/helm/helm/master/scripts/get | bash
+```
+
+Install helm tiller:
+
+```
+helm init
```

Comment:
Does the default service account of tiller have the right permission?

Reply:
yes, the tiller is deployed in kube-system namespace, default account of kube-system namespace has the permissions required by helm.